### PR TITLE
Use ElasticMQ instead of fake_sqs

### PR DESCRIPTION
**Pull request by akshaymankar**
https://wearezeta.atlassian.net/browse/WPB-5622

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

**Review by supersven — APPROVED** [review]
LGTM :+1:

**Comment by supersven** [comment]
Hm, Nix seems to be unhappy on CI. Probably, you should try to rebase and run the CI again...

**Comment by supersven** [comment]
> @supersven I had to change a few things, could you please review this again?

Sure. I'm taking a look. :eyeglasses:

File: libs/types-common-aws/src/Util/Test/SQS.hs [hunk 1]
```diff
@@ -54,17 +53,29 @@ data SQSWatcher a = SQSWatcher
...
+
+  ensureEmpty queueUrl
+  process <- async $ do
+    loop1 <- async $ recieveLoop queueUrl eventsRef
+    loop2 <- async $ recieveLoop queueUrl eventsRef
+    loop3 <- async $ recieveLoop queueUrl eventsRef
+    loop4 <- async $ recieveLoop queueUrl eventsRef
+    loop5 <- async $ recieveLoop queueUrl eventsRef
+    _ <- Async.waitAny [loop1, loop2, loop3, loop4, loop5]
+    throwIO $ BackgroundThreadNotRunning $ "One of the SQS recieve loops finished, all of them are supposed to run forever"
```
Comment (supersven) [inline libs/types-common-aws/src/Util/Test/SQS.hs:71]:
```suggestion
    throwIO $ BackgroundThreadNotRunning $ "One of the SQS receive loops finished, all of them are supposed to run forever"
```
Typo?

File: libs/types-common-aws/src/Util/Test/SQS.hs [hunk 2]
```diff
@@ -54,17 +53,29 @@ data SQSWatcher a = SQSWatcher
...
+
   pure $ SQSWatcher process eventsRef
   where
-    recieveLoop ref = do
+    recieveLoop queueUrl ref = do
```
Comment (supersven) [inline libs/types-common-aws/src/Util/Test/SQS.hs:75]:
```suggestion
    receiveLoop queueUrl ref = do
```

File: libs/types-common-aws/src/Util/Test/SQS.hs [hunk 3]
```diff
@@ -54,17 +53,29 @@ data SQSWatcher a = SQSWatcher
...
+  let queueUrl = view SQS.getQueueUrlResponse_queueUrl queueUrlRes
+
+  ensureEmpty queueUrl
+  process <- async $ do
+    loop1 <- async $ recieveLoop queueUrl eventsRef
+    loop2 <- async $ recieveLoop queueUrl eventsRef
+    loop3 <- async $ recieveLoop queueUrl eventsRef
+    loop4 <- async $ recieveLoop queueUrl eventsRef
+    loop5 <- async $ recieveLoop queueUrl eventsRef
+    _ <- Async.waitAny [loop1, loop2, loop3, loop4, loop5]
```
Comment (supersven) [inline libs/types-common-aws/src/Util/Test/SQS.hs:70]: Hm, somehow I don't get why you're calling `recieveLoop` five times here with the same parameters? :thinking: 
Maybe, leave a comment to explain this?

**Comment by akshaymankar** [comment]
Thanks @supersven I added comments to explain the choices and fixed the typo. Please let me know if this makes sense now.

File: libs/types-common-aws/src/Util/Test/SQS.hs (hunk 3 above)
Comment (akshaymankar) [reply libs/types-common-aws/src/Util/Test/SQS.hs]: I added a comment. Basically adding more loop increases the throughput of the consumer making tests not have to wait too long.

File: hack/helm_vars/fake-aws/values.yaml [hunk 4]
```diff
@@ -1,3 +1,29 @@
...
+  - "integration-brig-events-internal"
+  - "integration-gundeck-events"
+  - "integration-user-events.fifo"
+  - "integration-team-events.fifo"
+
+  - "integration-brig-events3"
+  - "integration-brig-events-internal3"
+  - "integration-gundeck-events3"
+  - "integration-user-events3.fifo"
+  - "integration-team-events3.fifo"
```
[the remark this answers is not in the document]
Comment (akshaymankar) [reply hack/helm_vars/fake-aws/values.yaml:24]: The number `2` backend runs in another namespace so it actually uses the queues without any suffix. 
I've added a comment to explain this.

**Review by supersven — APPROVED** [review]
LGTM :+1:

**Pull request merged**